### PR TITLE
feat(i18next): support interaction

### PR DESCRIPTION
**Pull request by Mzato0001**

### CODE REVIEW

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [inline packages/i18next/src/lib/functions.ts:3]: Hmm should I move `CommandInteraction` to the start?

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [inline packages/i18next/src/lib/functions.ts:3]: Should I move `CommandInteraction` to the start?

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:3]: They should be alphabetical, yes. Run prettier on your code and it'll handle that for you.

**Comment by Nytelife26** [comment]
If there's no extra logic required, then LGTM. I'll wait for Favna's or Kyra's input on that, though

**Review by favna — DISMISSED** [review]

**Review by favna — APPROVED** [review]

**Review by favna — CHANGES_REQUESTED** [review]
`editLocalized` and `replyLocalized` also still need to be updated

**Review by favna — DISMISSED** [review]

**Review by vladfrangu — COMMENTED** [review]
Does this support ContextMenu and Button Interactions?

**Review by kyranet — CHANGES_REQUESTED** [review]
If we're renaming `author` to `user`, the former should still work, otherwise this is a breaking change.

You can make a wrapper which adds an `author` getter with `process.emitWarning` to emit a deprecation warning. In next major, we shall remove the function and all of its references.

**Comment by Mzato0001** [comment]
> Does this support ContextMenu and Button Interactions?

It should support `ContextMenuInteraction` and `ButtonInteraction`, but for Buttons the type is a little different.
I'll add a commits to change it.

File: packages/i18next/src/lib/functions.ts
Comment (vladfrangu) [inline packages/i18next/src/lib/functions.ts:329]: This is not what kyra meant. If `author` is used, it should emit a warning and return whatever is present there, else the context.user

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:329]: well actually the `user` is just renamed from `author`

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:329]: the only difference is there is no `author` property from `Interaction` but I think its okay

**Comment by favna** [comment]
Oh yeah I added author back and added the TSDoc but I forgot to map it to user lol... Oops.

**Review by favna — DISMISSED** [review]

**Review by favna — APPROVED** [review]

File: packages/i18next/src/lib/functions.ts
Comment (vladfrangu) [inline packages/i18next/src/lib/functions.ts:218]: This is the exact same code

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:218]: About [this changes request](https://github.com/sapphiredev/plugins/pull/272/files/3d59b5b7208d3d7a485f7c0b385d83c3228d3da2#r851659889)?

Comment (vladfrangu) [reply packages/i18next/src/lib/functions.ts:218]: Plese check the code I wrote the review for...

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:218]: oh sorry

**Review by vladfrangu — DISMISSED** [review]

File: packages/i18next/src/lib/functions.ts
Comment (kyranet) [inline packages/i18next/src/lib/functions.ts:345]: Missing setter, the property is readonly at the moment, which makes it breaking.

And I guess it needs to be configurable and writable?

File: packages/i18next/src/lib/functions.ts
Comment (kyranet) [inline packages/i18next/src/lib/functions.ts:342]: You need to guard it with and `if` condition, right now accessing `context.author` will result on endless deprecation warnings, but it should be emitted only once.

**Review by kyranet — APPROVED** [review]

**Review by kyranet — CHANGES_REQUESTED** [review]
Wrong button. ^

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:342]: oh i see

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:37]: I do approve of the rename to user from author now since it makes more sense in a way, but the fact that it is no longer ordered by scale is a little annoying. Can we ignore our instincts in terms of alphabetical ordering and put user at the start here?

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:277]: This is somewhat unclear. Are you trying to say "an interaction that has been replied to"? If you clarify what this is supposed to mean, I can help you find the best way to express it.

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:280]:
```suggestion
 * @param options The language keys that will resolve to the new interaction content.
```

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:296]:
```suggestion
 * @param options An object of valid options, requiring at least a `keys` field.
```
I'd suggest linking to the valid options and maybe a `seealso` for the overload where the keys are described.

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:330]: I'd argue that the comments here don't really add much clarity, so they can be removed safely. After all, there are only two branches, both of which are quite self-descriptive.

File: packages/i18next/src/lib/types.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/types.ts:145]: Replace everything here of the form "The X for which the language should be fetched" (passive voice) to "The X to fetch the preferred language for"

**Review by Nytelife26 — CHANGES_REQUESTED** [review]

File: packages/i18next/src/lib/types.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/types.ts:140]:
```suggestion
	 * @deprecated Use {@link InternationalizationContext.user} instead; this will be removed in the next major version.
```

File: packages/i18next/src/lib/functions.ts
Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:277]: Hmm yes since deferred is also means as replied

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:277]: To defer means to postpone or cede via an acknowledgment of superiority.
For instance, I can defer a meeting, or I can defer to your programming knowledge, but you wouldn't talk about someone deferring a message (because that would mean delaying the message).

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:277]: In any case, assuming this follows what you mean, here's a good rephrasal.

```suggestion
 * Edits an interaction or reply to an interaction using the given language `keys`.
```

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:277]: Hmm I means this is the [`BaseCommandInteraction#deferReply`](https://discord.js.org/#/docs/discord.js/stable/class/BaseCommandInteraction?scrollTo=deferReply) method

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:277]: Oh nope its this one [`BaseCommandInteraction#deferred`](https://discord.js.org/#/docs/discord.js/stable/class/BaseCommandInteraction?scrollTo=deferred)

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:277]: My apologies. In that case, you appear to have misinterpreted Discord's use of the word "deferred". It is in fact a delay since they postpone the reply by showing a waiting message.

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:277]: Ic thanks

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:277]: My latest suggestion edit should capture the intended meaning correctly.

File: packages/i18next/src/lib/functions.ts
Comment (Nytelife26) [inline packages/i18next/src/lib/functions.ts:149]: I'd say keep the links in, but append `object` instead. For instance, here,
```
* Replies to another message using a {@link LocalizedMessageOptions} object
```
Remember to use "an" instead of "a" if the object name starts with a vowel sound.

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:149]: Keep the links and also the `see` tsdoc?

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:149]: like this?
image
or this?
image

Comment (Nytelife26) [reply packages/i18next/src/lib/functions.ts:149]: Now I think about it, actually, the links should be directly in the parameter description. The description of this specific function should be something like "Replies to a message using localization" and the description should contain
```ts
* @param options A {@link LocalizedMessageOptions} object, requiring at least a `keys` field.
```

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:149]: Hmm I think this should be fine
image

Comment (Mzato0001) [reply packages/i18next/src/lib/functions.ts:149]: hmm or
```ts
* Replies to another message using {@link LocalizedMessageOptions}.
```

File: packages/i18next/src/lib/functions.ts
Comment (kyranet) [inline packages/i18next/src/lib/functions.ts:13]:
```suggestion
import { deprecate } from 'node:util';
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by Nytelife26 — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**
